src/prefect/experimental/workers/base.py

```diff
@@ -36,6 +107,8 @@ def __bool__(self):
 @register_base_type
 class BaseWorker(abc.ABC):
     type: str
+    job_configuration: Optional[BaseJobConfiguration] = None
```

Comment:
The worker developer provides a class so the typing should reflect that
```suggestion
    job_configuration: Type[BaseJobConfiguration]
```